src/libs/actions/Task.js

```diff
@@ -0,0 +1,102 @@
+    let parentReport;
+
+    // If there isn't a parentReportId provided and assignee is given, we need to optimistically create a new parent chat report
+    // between the task creator and the assignee.
+    if (!parentReportID && assignee) {
+        parentReport = ReportUtils.getChatByParticipants([assignee]);
+    }
+
+    if (!parentReportID && !parentReport && assignee) {
+        parentReport = ReportUtils.buildOptimisticChatReport([assignee]);
+    }
+
+    // If parentReport is defined, use its reportID, otherwise, use the provided parentReportID
+    const finalParentReportID = parentReport ? parentReport.reportID : parentReportID;
+
+    // Open the parent report if assignee is given
+    if (assignee) {
+        Report.openReport(finalParentReportID, [assignee]);
+    }
+
+    const optimisticTaskReport = ReportUtils.buildOptimisticTaskReport(ownerEmail, assignee, finalParentReportID, parentReportActionID, name, description);
+    const optimisticCreatedAction = ReportUtils.buildOptimisticCreatedReportAction(ownerEmail);
+
+    // AddCommentReportAction on the parent chat report
+    const optimisticAddCommentReportAction = ReportUtils.buildOptimisticAddCommentReportAction(finalParentReportID, optimisticTaskReport.reportID);
+
+    const optimisticData = [
+        {
+            method: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${finalParentReportID}`,
+            value: optimisticCreatedAction,
+        },
+        {
+            method: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${finalParentReportID}`,
+            value: parentReport,
+        },
+        {
+            method: CONST.ONYX.METHOD.SET,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${optimisticTaskReport.reportID}`,
+            value: optimisticTaskReport,
+        },
+        optimisticAddCommentReportAction,
+    ];
+
+    const successData = {};
+
+    const failureData = {
+        undoActions: [
+            {
+                method: 'DELETE_TASK_REPORT',
+                reportID: optimisticTaskReport.reportID,
+            },
+            {
+                method: 'DELETE_REPORT_ACTION',
+                reportID: finalParentReportID,
+                actionID: optimisticCreatedAction.actionID,
+            },
+            {
+                method: 'DELETE_ADD_COMMENT_REPORT_ACTION',
+                reportID: finalParentReportID,
+                actionID: optimisticAddCommentReportAction.actionID,
+            },
+        ],
+    };
+
+    return API.write(
+        'CreateTask',
+        {
+            parentReportActionID,
+            parentReportID: finalParentReportID,
+            taskReportID,
+            name,
+            description,
+            assignee,
+            assigneeChatReportID,
+        },
+        {optimisticData, successData, failureData},
+    );
```

Comment:
Either here, or in another method CreateTaskAndNavigate let's navigate to the created task after we create it

Reply:
I don't think there's a scenario where we don't want to navigate to the created task, so I'm going to include it in this function.